### PR TITLE
[NO-ISSUE] Bulk speedup

### PR DESCRIPTION
**Pull request by blackheaven**
## Proposed changes

It's actually a draft since it has the following shortcomings:

- [ ] A memory leak (13.5 GB at the end)
- [ ] Very low parallelization (x6 on 20-core CPU)
- [ ] Outputs is interleaved

Results (on a 20-core CPU):
- `before`: 4745.19s user 916.19s system 144% cpu 57:59.78 total
- `after`: 6869.13s user 1557.62s system 619% cpu 22:39.85 total

## Contributor checklist

- [ ] My PR is related to \<insert ticket number> 
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/flora-pm/flora-server/blob/development/CONTRIBUTING.md)
- [ ] I have inserted my change and a link to this PR in the [CHANGELOG](https://github.com/flora-pm/flora-server/blob/development/CHANGELOG.md)

### CODE REVIEW

**Comment by tchoutri** [comment]
@Mergifyio rebase

File: flake.nix [hunk 1]
```diff
@@ -5,7 +5,7 @@
       url = "github:numtide/flake-utils";
     };
     horizon-platform = {
-      url = "git+https://gitlab.homotopic.tech/horizon/horizon-platform";
+      url = "github:blackheaven/horizon-platform-forked";
```
Comment (tchoutri) [inline flake.nix:8]: Have you opened a PR to upstream your changes? Or could you leave a comment on the nature of your changes?

**Comment by tchoutri** [comment]
@mergifyIO rebase

**Comment by tchoutri** [comment]
@blackheaven regarding interleaved output it's alright

**Comment by blackheaven** [comment]
I figured out it was already a defect.

It'll take a proper logger and an effect change to be fixed.

Regarding nix overrides/layers, there are incompatible derivations, I have to figure out how to override it locally.

**Comment by tchoutri** [comment]
@mergifyIO rebase

**Comment by tchoutri** [comment]
@mergifyIO rebase

**Comment by tchoutri** [comment]
@mergifyIO rebase

**Comment by tchoutri** [comment]
Good enough.

**Pull request merged**